new-frontiers-14/frontier-station-14
# NFSD Criminal Records Fix + Shipyard Fax Records System

## About the PR


It allows proper function of criminal records by adding a workaround which lets NFSD Outpost receive a record of each player that spawned in no matter in what location, can be Frontier Outpost, Latejoining vessel etc, it will get it without doubling.

This way NFSD Outpost has records of every single player, and criminal record console it is possible to write notes on player records as well as set their wanted level which works globally(from testing)

Adds system for Pirates and future antags. Where it can either fake the records, AKA real job won't be visible and replaced with one of the three most common jobs (contractor, pilot, mercenary), OR where they won't be registered in record at all.


On top of it is added the Shipyard Records, which every time someone purchases a vessel, a special fax will receive information and print an unique paper describing the vessel, captain and details regarding the captain as well as time.
This should drastically aid NFSD in their job as now they can gather and record evidence more easily.
On top of that vessels won't be anonymous anymore, and SR, STC and Sherrif can figure out who owns what ship.

Adds class to each vessel! Which can more precisely describe what kind of hsip and purpose certain vessel is.
The shipyard record paper color will match it's class.

Moreover Blackmarket and Syndicate vessels will have their information mixed with fake one. Giving detectives something to poke at instead of getting all information on silver platter.

=========
IMPORTANT ASPECT
This PR is gonna require mapping extra 2 faxes [vessel records] to Sherrif's and Station Represenative's offices.
Additionally NFSD Outpost will require extra mapped Criminal Records Consoles to be able to access those.

Also vessel Classes will need to be added to every single vessel in the map prototype, otherwise it will use the default "undetermined" type.

## Why / Balance

NFSD greatly lacks managing as well as intel gathering tools. This will mainly assist detectives in forensics. Sherrifs in writing more accurate warrants or even sending patrols to check on suspected ships.
Can be used to record infractions on vessel record themselves.
For STC it will be a MASSIVE help as they got essentially list of every single vessel that exists, able to easily write when they docked and when they should undock.
SR will be able to use this information for other functions as well.

## How to test

Add "AdditionalStationRecordsComponent" to NFSD Outpost
Latejoin onto station/ship that isnt NFSD outpost after its added
spawn criminal record console on NFSD
check if player information shows up

for shipyard fax records
spawn special fax with "ShipyardRecordPaperComponent" component
go to any shipyard console
purchase a ship
check the paper that this fax outputted

## Media

![obraz](https://github.com/user-attachments/assets/6e53230e-3dcc-4871-896d-dfeee5b90398)



## Breaking changes

Any changes to paper and faxes

**Changelog**

- fix: Fixed NFSD criminal as well as station records. Now they will contain information of every single play in the sector.
- tweak: Changed pirate's old records showing in NFSD's criminal records as their old jobs
- add: Added Shipyard Records Fax system. Which should greatly assist with vessel administration and forensics as each ship has described Captain to them
